ivadomed/loader/utils.py

```diff
@@ -501,3 +503,81 @@ def reorient_image(arr, slice_axis, nib_ref, nib_ref_canonical):
     trans_orient = nib.orientations.ornt_transform(ras_orientation, ref_orientation)
     # apply transformation
     return nib.orientations.apply_orientation(arr_ras, trans_orient)
+
+def create_bids_dataframe(loader_params, derivatives):
+    """Create a dataframe containing all BIDS image files in a bids_path and their metadata.
+
+    Args:
+        loader_params (dict): Loader parameters, see :doc:`configuration_file` for more details.
+        derivatives (bool): If True, derivatives are indexed.
+
+    Returns:
+        df (pd.DataFrame): Dataframe containing all BIDS image files indexed and their metadata.
+    """
+
+    # Get bids_path, bids_config, target_suffix and extensions from loader parameters
+    bids_path = loader_params['bids_path']
+    bids_config = loader_params['bids_config']
+    target_suffix = loader_params['target_suffix']
+    extensions = loader_params['extensions']
+
+    # Suppress a Future Warning from pybids about leading dot included in 'extension' from version 0.14.0
+    # The config_bids.json file used matches the future behavior
+    # TODO: when reaching version 0.14.0, remove the following line
+    pybids.config.set_option('extension_initial_dot', True)
+
+    # Initialize BIDSLayoutIndexer and BIDSLayout
+    # validate=True by default for both indexer and layout, BIDS-validator is not skipped
+    # Force index for subfolders starting with string "sub", and samples tsv and json files (for microscopy)
+    force_index = [d for d in os.listdir(bids_path) if (os.path.isdir(os.path.join(bids_path, d))) and (d.startswith("sub"))]
+    force_index.extend(['samples.tsv', 'samples.json'])
+    indexer = pybids.BIDSLayoutIndexer(force_index=force_index)
+    layout = pybids.BIDSLayout(bids_path, config=bids_config, indexer=indexer, derivatives=derivatives)
```

Comment:
I see we have a fork of pybids, do we plan on overwriting some of their code? If so, we might want to handle the error when .json files exist, but are empty (https://github.com/ivadomed/ivadomed/issues/532). Currently, we still have an error in this case.

Reply:
I was planning on making PR from the fork before I found the custom config file. For now, it seems easier to wait until BEP microscopy is merged before submitting changes to pybids. But the json file error has nothing to do with microscopy so yes, we could propose changes.